test/hotspot/jtreg/compiler/c2/TestDivModNodes.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2024 Red Hat and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.c2;
+
+import compiler.lib.ir_framework.*;
+import compiler.lib.ir_framework.Test;
+
+import java.util.Random;
+
+/*
+ * @test
+ * @bug 8332442
+ * @summary Test that DIV and MOD nodes are converted into DIVMOD where possible
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
```

Comment:
Is `C2` really required for this test? Or could another compiler also benefit from your test?

Reply:
Removed C2 requirements for validating arithmetic results on other compilers too